lib/upload_system_log.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+#Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package upload_system_log;
+# Summary:  base class for collecting and uploading system log
+# Maintainer: Yong Sun <yosun@suse.com>
+
+use strict;
+use base "Exporter";
+use Exporter;
+use testapi;
+use utils;
+use base "opensusebasetest";
+
+our @EXPORT = qw(upload_system_logs);
+
+
+# Save the output of $cmd into $file and upload it
+sub system_status {
+    my $log = shift || "/tmp/system-status.log";
```

Comment:
might be wrong , but isn't first param will be always $self ? so your function will never be able to read passed param.
 
```
my ($self, $log) = @;
$log //="/tmp/system-status.log";
```
this should work

Reply:
Not Perl expert, will take a try.